<launch>
  <include file="$(find ydlidar)/launch/ydlidar.launch" > </include>
  <!-- <include file="$(find robot_pose_ekf)/robot_pose_ekf.launch"> </include> -->
  <!-- <include file="$(find xf_mic_asr_offline)/launch/xf_mic_asr_offline.launch" > </include> -->
  <node name="map_server" pkg="map_server" type="map_server" args="$(find ucar_map)/maps/nav.yaml" output="screen">
   <param name="frame_id" value="map" />
  </node> 
  <include file="$(find ucar_nav)/launch/gear3_1/amcl/amcl_omni.launch" > </include>
  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
    <param name="base_global_planner" value="global_planner/GlobalPlanner"/>
    <param name="base_local_planner" value="teb_local_planner/TebLocalPlannerROS"/>
    <rosparam file="$(find ucar_nav)/launch/gear3/move_base/costmap_common_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find ucar_nav)/launch/gear3_1/move_base/costmap_common_params.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find ucar_nav)/launch/gear3_1/move_base/global_planner_params.yaml" command="load" />
    <rosparam file="$(find ucar_nav)/launch/gear3_1/move_base/teb_local_planner_params.yaml" command="load" />
    <rosparam file="$(find ucar_nav)/launch/gear3_1/move_base/local_costmap_params.yaml" command="load" />
    <rosparam file="$(find ucar_nav)/launch/gear3_1/move_base/global_costmap_params.yaml" command="load" />
    <rosparam file="$(find ucar_nav)/launch/gear3_1/move_base/move_base_params.yaml" command="load" />
  </node>
  <node name="qr_server" pkg="photo_and_qr" type="server.py" output="screen" /> 
  <node name="publish_base_link" pkg="ucar_controller" type="publish_base_link.py" output="screen" />
  <!-- <node name="nongoal" pkg="ucar_nav" type="nongoal.py" output="screen" /> -->
  <include file="$(find ucar_controller)/launch/base_driver.launch" > </include>
</launch>